libr/bin/p/bin_any.c

```diff
@@ -6,40 +6,38 @@
 #include <r_bin.h>
 #include <r_magic.h>
 
-static char *get_filetype(RBinFile *bf) {
-	ut8 buf[4096] = {
-		0
-	};
+static char *get_filetype(RBuffer *obj) {
+	ut8 buf[4096] = { 0 };
 	char *res = NULL;
 	RMagic *ck;
-	if (!bf) {
+	ck = r_magic_new (0);
+	if (!ck) {
 		return NULL;
 	}
-	ck = r_magic_new (0);
-	if (ck && bf && bf->buf) {
-		const char *tmp = NULL;
-		// TODO: dir.magic not honored here
-		char *pfx = r_str_newf (R_JOIN_2_PATHS ("%s", R2_SDB_MAGIC), r_sys_prefix (NULL));
-		r_magic_load (ck, R2_SDB_MAGIC);
-		r_buf_read_at (bf->buf, 0, buf, sizeof (buf));
-		tmp = r_magic_buffer (ck, buf, sizeof (buf));
-		if (tmp) {
-			res = strdup (tmp);
-		}
-		free (pfx);
+
+	const char *tmp = NULL;
+	// TODO: dir.magic not honored here
+	char *pfx = r_str_newf (R_JOIN_2_PATHS ("%s", R2_SDB_MAGIC), r_sys_prefix (NULL));
+	r_magic_load (ck, R2_SDB_MAGIC);
+	r_buf_read_at (obj, 0, buf, sizeof (buf));
+	tmp = r_magic_buffer (ck, buf, sizeof (buf));
+	if (tmp) {
+		res = strdup (tmp);
 	}
+	free (pfx);
 	r_magic_free (ck);
 	return res;
 }
 
 static RBinInfo *info(RBinFile *bf) {
+	RBuffer *any_obj = bf->o->bin_obj;
 	RBinInfo *ret = R_NEW0 (RBinInfo);
 	if (!ret) {
 		return NULL;
 	}
 	ret->lang = "";
 	ret->file = bf->file? strdup (bf->file): NULL;
-	ret->type = get_filetype (bf);
+	ret->type = get_filetype (any_obj);
```

Comment:
why call it any_obj instead of bin_obj

Reply:
dunno, it is the `any` plugin, so `any_obj` :D